app/src/main/java/com/example/task_planner_app/ui/fragments/CreateTaskFragment.kt

```diff
@@ -0,0 +1,121 @@
+            override fun onItemSelected(
+                parent: AdapterView<*>?,
+                view: View?,
+                position: Int,
+                id: Long
+            ) {
+                selectedStatus = statusList[position]
+            }
+
+            override fun onNothingSelected(parent: AdapterView<*>?) {}
+        }
+
+        binding.insertDateTask.setOnClickListener {
+            showDatePickerDialog()
+        }
+
+        binding.fabSaveTask.setOnClickListener {
+            val description = binding.insertDescriptionTask.text.toString()
+            val responsible = binding.insertResponsibleTask.text.toString()
+            val date = binding.insertDateTask.text.toString()
+            val status = binding.insertStatusTask.selectedItem.toString()
+            val createTaskValidation = validateCreateTaskFields(description, responsible, date, status)
+            if (createTaskValidation) {
+                binding.fabSaveTask.visibility = View.GONE
+                showLoading()
+                viewModel.createTask(description, responsible, date, status)
+                viewModel.successLiveData.observe(viewLifecycleOwner, { taskCreatedSuccessfuly ->
+                    if (taskCreatedSuccessfuly) {
+                        hideLoading()
+                        FancyToast.makeText(activity, "The task was created successfully",
+                            FancyToast.LENGTH_LONG, FancyToast.SUCCESS, true).show()
+                        goHomeFragmentFromCreateTaskFragment()
+                    }
+                })
+            } else {
+                FancyToast.makeText(activity, "Please, insert all fields",
+                    FancyToast.LENGTH_LONG, FancyToast.ERROR, true).show()
+            }
+        }
+
+        return binding.root
+    }
+
+    private fun hideLoading() {
+        loadingDialog?.let {
+            if (it.isShowing)
+                it.cancel()
+        }
+    }
+
+    private fun showLoading() {
+        hideLoading()
+        loadingDialog = Common.showLoadingDialog(requireContext())
+    }
+
+    private fun showDatePickerDialog() {
+        val datePicker = DatePickerFragment { day, month, year -> onDateSelected(day, month, year) }
+        activity?.let { datePicker.show(it.supportFragmentManager, "datePicker") }
+    }
+
+    fun onDateSelected(day: Int, month: Int, year:Int) {
+        binding.insertDateTask.setText("$year/$month/$day")
+    }
+
+    private fun goHomeFragmentFromCreateTaskFragment() {
+        findNavController().navigate(R.id.action_createTaskFragment_to_homeFragment)
+    }
+
+    private fun validateCreateTaskFields(description: String, responsible: String, date: String, status: String): Boolean {
+        if (description.isEmpty() || responsible.isEmpty() || date.isEmpty() || status.isEmpty()) {
+            FancyToast.makeText(activity, "Please, fill all fields", FancyToast.LENGTH_LONG, FancyToast.ERROR, true).show()
```

Comment:
the error message is not translable. This text should come from the Strings.xml file.